backend/corpora/common/utils/dropbox.py

```diff
@@ -0,0 +1,31 @@
+from urllib.parse import urlparse, parse_qs, urlencode, urlunparse
+
+import regex
+import requests
+
+
+def get_download_url_from_shared_link(url):
+    parsed_url = urlparse(url)
+    query = parse_qs(parsed_url.query)
+    params = {"dl": "1"}
+    query.update(params)
+
+    url_parts = list(parsed_url)
+    url_parts[4] = urlencode(query)
+    result = urlunparse(url_parts)
+    return result
+
+
+dropbox_link_rx = regex.compile(r"https://www.dropbox.com/s/[\w\d]+/.*\?dl=[01]")
```

Comment:
nice!!

Reply:
for those of us who are terrible at regex, would you mind including a comment about what this is doing?

Reply:
Really great point! +1 😄 

My stab is: 

Example Dropbox share link: https://www.dropbox.com/s/jnm2svlcr45blee/Get%20Started%20with%20Dropbox.pdf?dl=0

`[\w\d]+` matches 1 or more alphanumeric character -- `/jnm2svlcr45blee`
`.*` matches 0 or more any char, which takes care of encoded file name -- `/Get%20Started%20with%20Dropbox.pdf`
`\?dl=[01]` matches URL param `?dl=0` or `?dl=1` --  `?dl=0`

Reply:
added some test cases for the regex